### PR TITLE
Deps and workflow

### PR DESCRIPTION
**Pull request by piobab**
- bump deps
- remove `artifacts` workflow

### CODE REVIEW

File: Makefile.toml [hunk 1]
```diff
@@ -10,7 +10,7 @@ default_to_workspace = false
 # Directory with wasm files used by integration tests (another directory can be used instead, for example 'artifacts' from rust-optimizer)
 ARTIFACTS_DIR_PATH = "target/wasm32-unknown-unknown/release"
 # If you bump this version, verify RUST_VERSION correctness
-RUST_OPTIMIZER_VERSION = "0.12.13"
+RUST_OPTIMIZER_VERSION = "0.13.0"
```
Comment (larry0x) [inline Makefile.toml:13]: The latest version is `0.14.0`. Also the docker command has changed as of `0.13` (see: https://github.com/CosmWasm/rust-optimizer#usage), we should update:

```diff
 docker run --rm -v "$(pwd)":/code \
-  --mount type=volume,source="$(basename "$(pwd)")_cache",target=/code/target \
+  --mount type=volume,source="$(basename "$(pwd)")_cache",target=/target \
   --mount type=volume,source=registry_cache,target=/usr/local/cargo/registry \
   ${image}
```

**Review by larry0x — APPROVED** [review]

**Comment by larry0x** [comment]
I was about to push a commit to do some further cleanups (remove unused dependencies and features, format toml files...) but I'm going to leave it to a separate PR (so that the diff is easy to review)

File: Makefile.toml (hunk 1 above)
Comment (piobab) [reply Makefile.toml:13]: ahh thx, forgot about it. I am checking 0.14 because it has 1.71 rust version and it caused some problem on testing..

Comment (piobab) [reply Makefile.toml:13]: Leaving for now with 0.13. Still errors for unknown opcodes:
```
Trying to read wasm file: mars_oracle_osmosis
thread 'update_spot_with_different_pool' panicked at 'called `Result::unwrap()` on an `Err` value: ExecuteError { msg: "failed to execute message; message index: 0: Error calling the VM: Error during static Wasm validation: Wasm bytecode could not be deserialized. Deserialization error: \"Unknown opcode 194\": create wasm contract failed" }'
```

**Comment by piobab** [comment]
@larry0x I am merging. Feel free to create new PR.

**Pull request merged**